File: FieldOps.Domain/Entities/Shift.cs
namespace FieldOps.Domain.Entities;

public class Shift
{
    public int Id { get; set; }
    public int SiteId { get; set; }
    public string Name { get; set; } = string.Empty;
    public TimeOnly StartTime { get; set; }
    public TimeOnly EndTime { get; set; }
}

File: client/src/types/shift.ts
export interface Shift {
  id: number;
  siteId: number;
  siteName: string;
  name: string;
  startTime: string;
  endTime: string;
}

export interface CreateShiftRequest {
  siteId: number;
  name: string;
  startTime: string;
  endTime: string;
}

export interface ShiftAssignment {
  id: number;
  shiftId: number;
  shiftName: string;
  siteId: number;
  siteName: string;
  personnelId: number;
  personnelName: string;
  workDate: string;
  checkIn: string | null;
  checkOut: string | null;
  status: string;
}

export interface CreateShiftAssignmentRequest {
  shiftId: number;
  personnelId: number;
  workDate: string;
}
